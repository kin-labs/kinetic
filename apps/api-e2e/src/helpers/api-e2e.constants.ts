import { getAuthUsers } from '@kin-kinetic/api/config/data-access'
import { UserRole } from '@prisma/client'

const users = getAuthUsers(process.env.AUTH_USERS)
const { username, password } = users.find((user) => user.role === UserRole.Admin)

export const ADMIN_USERNAME = username
export const ADMIN_PASSWORD = password
