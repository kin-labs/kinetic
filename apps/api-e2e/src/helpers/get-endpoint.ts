import { INestApplication } from '@nestjs/common'
import * as request from 'supertest'

export function getEndpoint(app: INestApplication, endpoint: string) {
  return request(app.getHttpServer()).get(endpoint)
}
