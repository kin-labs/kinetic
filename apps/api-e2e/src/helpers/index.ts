export * from './api-e2e.constants'
export * from './get-endpoint'
export * from './initialize-e2e-app'
export * from './run-query'
export * from './random-int'
