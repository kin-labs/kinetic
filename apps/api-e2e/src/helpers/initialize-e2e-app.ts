import { INestApplication, ValidationPipe } from '@nestjs/common'
import { Test, TestingModule } from '@nestjs/testing'
import { AppModule } from '@kin-kinetic/api-app-module'

export async function initializeE2eApp(): Promise<INestApplication> {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile()

  const app = moduleFixture.createNestApplication()
  app.useGlobalPipes(new ValidationPipe({ transform: true }))
  app.setGlobalPrefix('api')
  await app.init()
  return app
}
