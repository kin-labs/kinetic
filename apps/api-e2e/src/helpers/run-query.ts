import { INestApplication } from '@nestjs/common'
import { UserRole } from '@prisma/client'
import { ASTNode } from 'graphql'
import { print } from 'graphql/language/printer'
import * as request from 'supertest'
import { Login, LoginInput } from '../generated/api-sdk'
import { ADMIN_PASSWORD } from './api-e2e.constants'

export function runGraphQLQuery(
  app: INestApplication,
  query: ASTNode,
  variables?: Record<string, unknown>,
  { token }: { token?: string } = {},
) {
  return request(app.getHttpServer())
    .post('/graphql')
    .set('Authorization', token ? `Bearer ${token}` : null)
    .send({ query: print(query), variables: variables })
}

export function runGraphQLQueryAdmin(
  app: INestApplication,
  token: string,
  query: ASTNode,
  variables?: Record<string, unknown>,
) {
  return runGraphQLQuery(app, query, variables, { token })
}

export function runLoginQuery(app: INestApplication, email: string, role: UserRole) {
  const input: LoginInput = { password: ADMIN_PASSWORD, email }
  return runGraphQLQuery(app, Login, { input }).expect(200)
}
