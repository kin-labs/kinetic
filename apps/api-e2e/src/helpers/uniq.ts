import { getRandomInt } from './random-int'

export function uniq(prefix: string) {
  return `${prefix}${uniqInt()}`
}
export function uniqInt(): number {
  return Math.floor(Math.random() * 10000000)
}
export function randomAppIndex(): number {
  return getRandomInt(1_000, 1_000_000)
}
