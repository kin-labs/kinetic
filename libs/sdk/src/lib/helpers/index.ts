export * from './get-solana-rpc-endpoint'
export * from './kin-to-quarks'
export * from './parse-kinetic-sdk-config'
export * from './parse-kinetic-sdk-endpoint'
