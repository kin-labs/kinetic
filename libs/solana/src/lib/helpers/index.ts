export * from './get-public-key'
export * from './parse-endpoint'
export * from './deserialize-transaction'
