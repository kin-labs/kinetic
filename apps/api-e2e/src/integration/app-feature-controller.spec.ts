import { INestApplication } from '@nestjs/common'
import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
import { Keypair } from '@solana/web3.js'
import { getEndpoint, initializeE2eApp, postEndpoint } from '../helpers'

describe('AppFeatureController (e2e)', () => {
  let app: INestApplication

  beforeEach(async () => {
    app = await initializeE2eApp()
  })

  afterEach(async () => {
    return app.close()
  })

  it('/api/app/devnet/1/config (GET)', () => {
    const feePayer = Keypair.fromSecretKey(Uint8Array.from(JSON.parse(process.env.APP_1_FEE_PAYER_BYTE_ARRAY)))
    return getEndpoint(app, '/api/app/devnet/1/config')
      .expect(200)
      .then((res) => {
        expect(res.body.app.index).toEqual(1)
        expect(res.body.app.name).toEqual('App 1')
        expect(res.body.mint.feePayer).toEqual(feePayer.publicKey.toBase58())
        expect(res.body.mint.programId).toEqual(TOKEN_PROGRAM_ID?.toBase58())
        expect(res.body.mint.publicKey).toEqual(process.env.DEFAULT_MINT_PUBLIC_KEY)
      })
  })

  it('Should not receive Incoming App Event Webhooks by default', () => {
    const payload = { foreignKey: 'some-foreign-key' }
    const headers = { authorization: 'Bearer Test' }
    return postEndpoint(app, '/api/app/devnet/1/webhook/event', payload, headers)
      .expect(400)
      .then((res) => expect(res.error).toMatchSnapshot())
  })

  it('Should not receive Incoming App Verify Webhooks by default', () => {
    const payload = { foreignKey: 'some-foreign-key' }
    const headers = { authorization: 'Bearer Test' }
    return postEndpoint(app, '/api/app/devnet/1/webhook/verify', payload, headers)
      .expect(400)
      .then((res) => expect(res.error).toMatchSnapshot())
  })

  it('Should not receive unknown Incoming App Webhooks ', () => {
    return postEndpoint(app, '/api/app/devnet/1/webhook/unknown')
      .expect(400)
      .then((res) => {
        expect(res.body).toMatchSnapshot()
      })
  })
})
