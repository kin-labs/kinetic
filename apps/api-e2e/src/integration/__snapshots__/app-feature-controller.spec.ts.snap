// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFeatureController (e2e) Should not receive Incoming App Event Webhooks by default 1`] = `[Error: cannot POST /api/app/1/webhook/event (400)]`;

exports[`AppFeatureController (e2e) Should not receive Incoming App Verify Webhooks by default 1`] = `[Error: cannot POST /api/app/1/webhook/verify (400)]`;

exports[`AppFeatureController (e2e) Should not receive unknown Incoming App Webhooks  1`] = `
Object {
  "message": "Unknown AppWebhookType",
  "name": "BadRequestException",
  "response": Object {
    "error": "Bad Request",
    "message": "Unknown AppWebhookType",
    "statusCode": 400,
  },
  "status": 400,
}
`;
